pkg/roachprod/vm/gce/gcloud.go

```diff
@@ -400,6 +400,30 @@ func (p *Provider) GetHostErrorVMs(
 	return hostErrorVMs, nil
 }
 
+// GetVMSpecs returns a json list of VM specs
+func (p *Provider) GetVMSpecs(l *logger.Logger, vms vm.List) ([]map[string]interface{}, error) {
+	if p.GetProject() == "" {
+		return nil, errors.New("project name cannot be empty")
+	}
+	if vms == nil {
+		return nil, errors.New("vms cannot be nil")
+	}
+	// Extract the spec of all VMs.
+	var vmSpecs []map[string]interface{}
+	for _, vmInstance := range vms {
+		var vmSpec map[string]interface{}
+		vmFullResourceName := "projects/" + p.GetProject() + "/zones/" + vmInstance.Zone + "/instances/" + vmInstance.Name
+		args := []string{"compute", "instances", "describe", vmFullResourceName, "--format=json"}
```

Comment:
Usability note, the output this spits out is **massive**. It looks like we can query for just certain fields via the `--format` field.

>To output only a set of fields from the available information, specify it using the '--format' flag:
>
> gcloud compute instances describe test-instance --format="yaml(name,status,disks)"

Might be worth figuring out which fields we care about and only logging those.

Reply:
Hmm actually, I think the command had a transient error and it appended a bunch of junk to the file for some reason. Tried it again a few times and the files are reasonable sizes, probably can ignore my previous comment 😄